src/microbe_stage/Patch.cs

```diff
@@ -188,15 +214,11 @@ public object Clone()
         var result = new PatchSnapshot
         {
             TimePeriod = TimePeriod,
-            SpeciesInPatch = new Dictionary<Species, long>(SpeciesInPatch.Count),
+            SpeciesInPatch = new Dictionary<Species, long>(SpeciesInPatch),
+            SpeciesGoneDefinitelyExtinct = new List<Species>(SpeciesGoneDefinitelyExtinct),
             Biome = (BiomeConditions)Biome.Clone(),
         };
 
-        foreach (var entry in SpeciesInPatch)
-        {
-            result.SpeciesInPatch.Add(entry.Key, entry.Value);
-        }
-
```

Comment:
Cloning species is starting to sound a bit expensive...
I already in some recent PR said that cloning a species object should be avoided, unless there's no other way to do something.

Reply:
As said above, cloning the species is necessary to get a solution with global population evaluation. I think recording the species that wen extinct at this round, although less elegant, is not necessarily worse.

However, this brings the benefit of not having to deal with a list to check, although it should stay rather small in size, given that not too many species will go extinct most times, and that snapshot are used few times anyways...

Reply:
Regarding the species cloning problem, I like the separate species data class solution as suggested by hhyyrylainen. Maybe a `MicrobeSpeciesInfo` class defined inside Patch (or on its own) where it will be an instance member of PatchSnapshot of type Dictionary called `RecordedSpeciesInfo`. It could be instantiated and populated from every `SpeciesInPatch` on snapshot creation.